### PR TITLE
Instrumenting special assert  in CPROVER_START after call into the function under test. (#398)

### PR DESCRIPTION
**Pull request by Degiorgio**

### CODE REVIEW

File: src/goto-instrument/cover.cpp [hunk 1]
```diff
@@ -1461,6 +1461,31 @@ bool instrument_cover_goals(
...
+  if(cmdline.isset("cover-traces-must-terminate"))
+  {
+    // instrument a 'fake' goal in CPROVER_START. This will rephrase
+    // the reachability problem  by asking BMC to provide a solution that
+    // satisfies a goal while getting to the end of the program-under-test.
+    const auto sf_it=
+      goto_functions.function_map.find(goto_functions.entry_point());
+    if(sf_it==goto_functions.function_map.end())
+    {
+      msg.error() << "cover-traces-must-terminate: invalid entry point"
```
Comment (mgudemann) [inline src/goto-instrument/cover.cpp]: please also emit the content of `entry_point` in the case of an error to facilitate debugging

File: src/goto-instrument/cover.cpp [hunk 2]
```diff
@@ -1461,6 +1461,31 @@ bool instrument_cover_goals(
...
+        << messaget::eom;
+      return true;
+    }
+    auto if_it=sf_it->second.body.instructions.end();
+    while(!if_it->is_function_call())
+      if_it--;
+    if_it++;
+    const std::string &comment="fake goal, reachability constraint.";
```
Comment (mgudemann) [inline src/goto-instrument/cover.cpp]: I'd prefer not calling it "fake", but maybe "Additional goal to ensure complete trace coverage"

**Comment by Degiorgio** [comment]
@mgudemann  done.

**Comment by Degiorgio** [comment]
@allredj, this PR is related to https://github.com/diffblue/cbmc/pull/1018 where  there are  regression tests.  Agreed regarding squashing and correct formatting of commit message.

**Comment by Degiorgio** [comment]
@allredj  done

**Pull request merged**